src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/HotReload/ProjectHotReloadAgent.cs

```diff
@@ -49,24 +50,23 @@ public ProjectHotReloadAgent(
     public IProjectHotReloadSession CreateHotReloadSession(
         string id,
         int variant,
-        string runtimeVersion,
+        string? runtimeVersion, // ignored
         ConfiguredProject configuredProject,
-        IProjectHotReloadLaunchProvider launchProvider,
-        IProjectHotReloadBuildManager buildManager,
+        IProjectHotReloadLaunchProvider? launchProvider, // ignored
+        IProjectHotReloadBuildManager? buildManager, // ignored
         IProjectHotReloadSessionCallback callback,
         ILaunchProfile launchProfile,
         DebugLaunchOptions debugLaunchOptions)
     {
         return new ProjectHotReloadSession(
             name: id,
             variant: variant,
-            runtimeVersion: runtimeVersion,
             hotReloadAgentManagerClient: _hotReloadAgentManagerClient,
             hotReloadOutputService: _hotReloadDiagnosticOutputService,
             deltaApplierCreator: _managedDeltaApplierCreator,
             callback: callback,
-            buildManager: buildManager,
-            launchProvider: launchProvider,
+            buildManager: configuredProject.Services.ExportProvider.GetExportedValueOrDefault<IProjectHotReloadBuildManager>(),
```

Comment:
You can MEF import `IActiveConfiguredValue<IProjectHotReloadBuildManager>` from constructor.
Using IActiveConfiguredValue enables us to inject mock object for unit tests